cobra/core/summary/metabolite_summary.py

```diff
@@ -55,106 +76,50 @@ def _generate(self):
             The DataFrame of flux summary data.
 
         """
-        if self.names:
-            emit = attrgetter('name')
-        else:
-            emit = attrgetter('id')
-
-        if self.solution is None:
-            self.met.model.slim_optimize(error_value=None)
-            self.solution = get_solution(self.met.model,
-                                         reactions=self.met.reactions)
-
-        rxns = sorted(self.met.reactions, key=attrgetter('id'))
-
-        data = [(emit(rxn), self.solution[rxn.id] * rxn.metabolites[self.met],
-                 rxn.build_reaction_string(use_metabolite_names=self.names),
-                 rxn) for rxn in rxns]
-
-        flux_summary = pd.DataFrame.from_records(
+        rxns = sorted(self.metabolite.reactions, key=attrgetter('id'))
+        data = [
+            (r.id, self.solution[r.id] * r.metabolites[self.metabolite],
+             r.metabolites[self.metabolite])
+            for r in rxns
+        ]
+        flux_summary = DataFrame(
             data=data,
-            index=[rxn.id for rxn in rxns],
```

Comment:
You could possibly declare the `index` here since you anyway set the `reaction` as index later. Also, then you could remove the `r.id` from `data`.